### PR TITLE
Clean up libgalois algorithms and add TODOs for notable issues.

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

File: libgalois/src/analytics/random_walks/random_walks.cpp [hunk 1]
```diff
@@ -485,6 +481,12 @@ RandomWalksWithWrap(katana::PropertyGraph* pg, RandomWalksPlan plan) {
     return res.error();
   }
 
+  // TODO(amp): This is incorrect. For Node2vec this needs to be:
+  //    Algorithm::Graph::Make(pg, {}, {}) // Ignoring all properties.
+  //  For Edge2vec this needs to be:
+  //    Algorithm::Graph::Make(pg, {}, {edge_type_property_name})
+  //  The current version requires the input to have exactly the properties
+  //  expected by the algorithm implementation.
```
Comment (arthurp) [inline libgalois/src/analytics/random_walks/random_walks.cpp:491]: This issue needs to be addressed to wrap `RandomWalks`. The algorithm will currently not work.

**Comment by arthurp** [comment]
There are a couple of smaller issues that I would like to address before wrapping:

https://github.com/KatanaGraph/katana/blob/e62ff0647e06e4511e31f73a1e89442604b0b5b1/libgalois/include/katana/analytics/subgraph_extraction/subgraph_extraction.h#L29-L35

https://github.com/KatanaGraph/katana/blob/e62ff0647e06e4511e31f73a1e89442604b0b5b1/libgalois/include/katana/analytics/random_walks/random_walks.h#L75-L80

I can convert all these into Github and/or JIRA issues if that's better for you. This was just a convenient way to communicate this for me at the moment.

**Comment by arthurp** [comment]
There are other TODOs in this PR, but they are mostly documentation issues, so they do not directly affect wrapping.

File: libgalois/src/analytics/random_walks/random_walks.cpp (hunk 1 above)
Comment (gurbinder533) [reply libgalois/src/analytics/random_walks/random_walks.cpp:491]: Good point. We need to separate them out into different applications.

**Comment by arthurp** [comment]
Closing this in favor of #177. This commit is in that PR and some of the issues are addressed there in one way or another.

**Pull request closed without merging**